src/shared/pam-util.c

```diff
@@ -95,10 +101,17 @@ int pam_acquire_bus_connection(pam_handle_t *handle, sd_bus **ret) {
         return PAM_SUCCESS;
 }
 
-int pam_release_bus_connection(pam_handle_t *handle) {
+int pam_release_bus_connection(pam_handle_t *handle, const char *module_name) {
+        _cleanup_free_ char *cache_id = NULL;
         int r;
 
-        r = pam_set_data(handle, "systemd-system-bus", NULL, NULL);
+        assert(module_name);
+
+        cache_id = strjoin("system-bus-", module_name);
+        if (!cache_id)
+                return pam_syslog_pam_error(handle, LOG_ERR, SYNTHETIC_ERRNO(ENOMEM), "Failed to allocate memory.");
```

Comment:
just use pam_log_oom()

Reply:
I even search for something like that, and missed it, I must be going blind